obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+                else:
+                    peri = ' '
+                    peri_round = False
+                # Extract amplitude and convert units
+                if amplitude.generic_amplitude:
+                    amp = amplitude.generic_amplitude
+                    if amplitude.unit in ['m', 'm/s', 'm/(s*s)', 'm*s']:
+                        amp *= 10**9
+                    # Otherwise we will assume that the amplitude is in counts
+                else:
+                    amp = np.nan
+                coda = ' '
+                if amplitude.magnitude_hint == 'Ml':
+                    phase_hint = 'IAML'
+                    impulsivity = ' '
+            else:
+                coda = int(amplitude.generic_amplitude)
+                peri = ' '
+                peri_round = False
+                amp = np.nan
+        else:
+            peri = ' '
+            peri_round = False
+            amp = np.nan
+            coda = ' '
+        # If the weight is 0 then we don't need to print it
+        if weight == 0 or weight == '0':
+            weight = 999  # this will return an empty string using _str_conv
+        # Generate a print string and attach it to the list
+        channel_code = pick.waveform_id.channel_code or '   '
+        pick_strings.append(' ' + pick.waveform_id.station_code.ljust(5) +
+                            channel_code[0] + channel_code[-1] +
+                            ' ' + impulsivity + phase_hint.ljust(4) +
+                            _str_conv(int(weight)).rjust(1) + ' ' +
+                            polarity.rjust(1) + ' ' +
+                            str(pick.time.hour).rjust(2) +
+                            str(pick.time.minute).rjust(2) +
+                            str(pick.time.second).rjust(3) + '.' +
+                            str(float(pick.time.microsecond) /
+                            (10 ** 4)).split('.')[0].zfill(2) +
+                            _str_conv(coda).rjust(5)[0:5] +
+                            _str_conv(round(amp, 1)).rjust(7)[0:7] +
+                            _str_conv(peri, rounded=peri_round).rjust(5) +
+                            _str_conv(azimuth).rjust(6) +
+                            _str_conv(velocity).rjust(5) +
+                            _str_conv(' ').rjust(4) +
+                            _str_conv(azimuthres).rjust(3) +
+                            _str_conv(timeres, rounded=2).rjust(5)[0:5] +
+                            _str_conv(' ').rjust(2) +
+                            distance.rjust(5) +
+                            _str_conv(caz).rjust(4) + ' ')
+        # Note that currently finalweight is unsupported, nor is velocity, or
+        # angle of incidence.  This is because obspy.event stores slowness in
+        # s/deg and takeoff angle, which would require computation from the
+        # values stored in seisan.  Multiple weights are also not supported in
+        # Obspy.event
+    return pick_strings
+
+
+def stationtoseisan(station):
```

Comment:
Good point, I will have a think - there is not much information in there, but it could be used - this is more output from a `Station` class though - might look at adding it as a plugin for `Inventory.write`.